### PR TITLE
Passer Tactic Updates

### PR DESCRIPTION
**Pull request by garethellis0**
**PLEASE REVIEW AFTER #557**, this PR includes all the changes from that one because dependencies

### Description
Updated Passer Tactic to properly move to behind the ball while waiting for the pass to start.

### Testing Done
Unit tests, tested on robots.

### Resolved Issues
Part of #373

### Length Justification
NA

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Start of document comments**: each `.cpp` and `.h` file should have a comment at the start of it. See files in the `thunderbots/software/geom` folder for examples.
- [x] **Function comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the classes defined in `thunderbots/software/geom`
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/thunderbots/software/ai/hl/stp/tactic/passer_tactic.cpp
Comment (MathewMacDougall) [inline src/thunderbots/software/ai/hl/stp/tactic/passer_tactic.cpp]: More of a note here (doesn't _have_ to be changed now), I think we should make `BALL_MAX_RADIUS_METERS` -> `2 * BALL_MAX_RADIUS_METERS`.

This is because it's very likely we will use this during free-kicks or set plays. In this case we have to be very careful about touching or moving the ball before we are truly ready to kick, otherwise the ball will be in-play and the other team may move to steal the ball or block it before we get the change to kick.

We saw on the field that the robot would sometimes touch the ball slightly and move it a bit. That will make the ball "in play" in a real game. By making the point slightly further from the ball we should be able to avoid touching it while still being really close.

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Only 1 minor comment. Otherwise LGTM.

File: src/thunderbots/software/ai/hl/stp/tactic/passer_tactic.cpp
Comment (garethellis0) [reply src/thunderbots/software/ai/hl/stp/tactic/passer_tactic.cpp]: Great point, changed.

**Comment by MathewMacDougall** [comment]
LGTM but will wait until #557 is merged and this is rebased

**Review by MathewMacDougall — DISMISSED** [review]
LGTM

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Pull request merged**
